satpy/readers/mirs_l2_nc.py

```diff
@@ -0,0 +1,506 @@
+            new_coords = self.nc.coords.keys()
+        return new_coords
+
+    @property
+    def platform_shortname(self):
+        """Get platform shortname."""
+        return self.filename_info['platform_shortname']
+
+    @property
+    def _get_platform_name(self):
+        """Get platform name."""
+        try:
+            res = PLATFORMS[self.filename_info['platform_shortname'].lower()]
+        except KeyError:
+            res = "mirs"
+        return res.lower()
+
+    @property
+    def _get_sensor(self):
+        """Get sensor."""
+        try:
+            res = SENSOR[self.filename_info["platform_shortname"].lower()]
+        except KeyError:
+            res = self.sensor_names
+        return res
+
+    @property
+    def sensor_names(self):
+        """Return standard sensor names for the file's data."""
+        return list(set(SENSOR.values()))
+
+    @property
+    def start_time(self):
+        """Get start time."""
+        # old file format
+        if self.filename_info.get("date", False):
+            s_time = datetime.datetime.combine(
+                self.force_date("date"),
+                self.force_time("start_time")
+            )
+            self.filename_info["start_time"] = s_time
+        return self.filename_info["start_time"]
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        # old file format
+        if self.filename_info.get("date", False):
+            end_time = datetime.datetime.combine(
+                self.force_date("date"),
+                self.force_time("end_time")
+            )
+            self.filename_info["end_time"] = end_time
+        return self.filename_info["end_time"]
+
+    def force_date(self, key):
+        """Force datetime.date for combine."""
+        if isinstance(self.filename_info[key], datetime.datetime):
+            return self.filename_info[key].date()
+        else:
+            return self.filename_info[key]
+
+    def force_time(self, key):
+        """Force datetime.time for combine."""
+        if isinstance(self.filename_info.get(key), datetime.datetime):
+            return self.filename_info.get(key).time()
+        else:
+            return self.filename_info.get(key)
+
+    def apply_atms_limb_correction(self, datasets, channel_idx, dmean, coeffs, amean, nchx, nchanx):
+        """Apply the atms limb correction to the brightness temperature data."""
+        datasets = datasets.persist()
+        ds = datasets[channel_idx]
+        new_ds = np.zeros(ds.shape, dtype=ds.dtype)
+        for fov_idx in range(N_FOV):
+            new_ds[:, fov_idx] = da.map_blocks(self.coeff_cums_calc,
+                                               datasets, channel_idx,
+                                               fov_idx, dmean, coeffs,
+                                               amean, nchx,
+                                               nchanx, dtype=ds.dtype)
+
+        return new_ds
+
+    @staticmethod
```

Comment:
There are some cases (multi-process scheduler) where dask doesn't always work well with functions that aren't globally defined. I'm not sure if staticmethods have this problem, but what are your thoughts on moving this staticmethod (and any other delayed/map_blocks'd methods) to be functions in the global scope?